js/tuxexchange.js

```diff
@@ -0,0 +1,567 @@
+        const market = this.findMarket (marketId);
+        // ALGO: all the values here check out, but there are a lot of undefineds
+        return {
+            'symbol': market['symbol'],
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'close': this.safeFloat (tickerData, 'last'),
+            'last': this.safeFloat (tickerData, 'last'),
+            'high': this.safeFloat (tickerData, 'high24hr'),
+            'low': this.safeFloat (tickerData, 'low24hr'),
+            'percentage': this.safeFloat (tickerData, 'percentChange'),
+            'baseVolume': this.safeFloat (tickerData, 'baseVolume'),
+            'quoteVolume': this.safeFloat (tickerData, 'quoteVolume'),
+            'bid': this.safeFloat (tickerData, 'highestBid'),
+            'ask': this.safeFloat (tickerData, 'lowestAsk'),
+            'bidVolume': undefined,
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'previousClose': undefined,
+            'change': undefined,
+            'average': undefined,
+            'info': tickerData,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchOrderBook () requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const codes = this.getIdsFromSymbol (symbol);
+        if (codes.base !== 'BTC') {
+            throw new NotSupported (this.id + ' this exchange only trades on symbols with BTC as base');
+        }
+        if (limit !== undefined) {
+            throw new NotSupported (this.id + ' fetchOrderBook () does not support a "limit" argument for this exchange');
+        }
+        let orderBookRequest = {
+            'coin': this.currencyId (codes.quote),
+            'method': 'getorders',
+        };
+        const response = await this.publicGet (orderBookRequest);
+        const orderBook = this.parseJson (response);
+        const result = this.parseOrderBook (orderBook);
+        return result;
+    }
+
+    parseTrade (trade, market) {
+        // The API exposees different fields for the gettrades and getmytradehistory endpoints
+        const isPrivateTrade = trade['orderId'] !== undefined;
+        const tradeDate = this.parseDate (this.safeString (trade, 'date'));
+        const timestamp = tradeDate.valueOf ();
+        let priceString = undefined;
+        if (isPrivateTrade) {
+            priceString = this.safeString (trade, 'price');
+        } else {
+            priceString = this.safeString (trade, 'rate');
+        }
+        const parsedTrade = {
+            'id': this.safeString (trade, 'tradeid'),
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': market.id,
+            'type': 'limit',
```

Comment:
I would leave this as undefined since you're not sure if they don't specify if they're trades from limit orders or market orders

Reply:
the exchange only allows limit orders (which I enforce below but had an incorrect error message so I fixed that). Seems like since I enforce orders are "limit" going in that they can be hardcoded coming out no?

Reply:
Oh wow, good call, that's fine then.  It's bizarre that they don't have market orders!